File: php/fixtures/texttest_fixture.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/../vendor/autoload.php';

use GildedRose\GildedRose;
use GildedRose\Item;

echo 'TextTest' . PHP_EOL;

$items = [
    new Item('Vest', 10, 20),
    new Item('Aged Brie', 2, 0),
    new Item('Juice', 5, 7),
    new Item('Diamond', 0, 80),
    new Item('Diamond', -1, 80),
    new Item('Backstage passes', 15, 20),
    new Item('Backstage passes', 10, 49),
    new Item('Backstage passes', 5, 49),
    // this conjured item does not work properly yet
    new Item('Bio Cake', 3, 6),
];

$app = new GildedRose($items);

$days = 2;
if ((is_countable($argv) ? count($argv) : 0) > 1) {
    $days = (int) $argv[1];
}

for ($i = 0; $i < $days; $i++) {
    echo "-------- day ${i} --------" . PHP_EOL;
    echo 'name, sellIn, quality' . PHP_EOL;
    foreach ($items as $item) {
        echo $item . PHP_EOL;
    }
    echo PHP_EOL;
    $app->updateQuality();
}

File: Kotlin/src/test/kotlin/com/gildedrose/TexttestFixture.kt
package com.gildedrose

fun main(args: Array<String>) {

    println("TextTest")

    val items = listOf(Item("Vest", 10, 20), //
            Item("Aged Brie", 2, 0), //
            Item("Juice", 5, 7), //
            Item("Diamond", 0, 80), //
            Item("Diamond", -1, 80),
            Item("Backstage passes", 15, 20),
            Item("Backstage passes", 10, 49),
            Item("Backstage passes", 5, 49),
            // this conjured item does not work properly yet
            Item("Bio Cake", 3, 6))

    val app = GildedRose(items)

    var days = 2
    if (args.size > 0) {
        days = Integer.parseInt(args[0]) + 1
    }

    for (i in 0..days - 1) {
        println("-------- day $i --------")
        println("name, sellIn, quality")
        for (item in items) {
            println(item)
        }
        println()
        app.updateQuality()
    }
}
